### PR TITLE
Create data model and transformers for Organization type

### PR DESCRIPTION
**Pull request by grahamlangford**
## What does this PR do?

- Part of #9131

## Discussion

- Needed to reuse the organization transformer in the service worker
- Realized the transformer belonged in the data layer
- Tried to follow the existing pattern to fully implement the transformer

For more information on our expectations for the PR process, see the
[code review principles doc](https://www.notion.so/pixiebrix/Code-Review-Principles-1ce7276b82a84d2a995d55ad85e1310d?pvs=4)

### CODE REVIEW

File: src/types/modTypes.ts [hunk 1]
```diff
@@ -46,7 +46,7 @@ export type SharingType =
 export type SharingSource = {
   type: SharingType;
   label: string;
-  organization?: Nullishable<Organization>;
+  organization?: Nullishable<AuthUserOrganization>;
```
Comment (grahamlangford) [inline src/types/modTypes.ts:49]: This was just the wrong type. Fixing the types revealed that

File: src/components/fields/schemaFields/widgets/DatabaseCreateModal.tsx [hunk 2]
```diff
@@ -70,10 +71,19 @@ const initialValues: DatabaseConfig = {
...
+    .filter(
+      (organization) =>
+        organization.members?.some(
+          (member) =>
+            // If the current user is an admin of the organization, then all of the members are listed for the organization
+            // Otherwise, only the current user is listed for the organization
+            // So if any listed member is an admin, the current user is an admin
+            member.role === UserRole.admin,
+        ),
+    )
```
Comment (grahamlangford) [inline src/components/fields/schemaFields/widgets/DatabaseCreateModal.tsx:83]: Removes the need for the lifted role. See:

File: src/types/contract.ts [hunk 3]
```diff
@@ -40,21 +40,6 @@ import {
...
-  manager = 5,
-}
-
-export type Organization = components["schemas"]["Organization"] & {
-  // The `role` property is added in the Redux RTK definition for getOrganizations (see api.ts)
-  // WARNING: currently this role is only accurate for Admin. All other users are passed as Restricted even if they have
-  // a Member or Developer role on the team
-  role: UserRole;
-};
-
```
Comment (grahamlangford) [inline src/types/contract.ts:57]: This type was never useful/accurate. See: https://github.com/pixiebrix/pixiebrix-extension/pull/9152/files#r1759349672

File: src/data/model/OrganizationMembers.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+};
+
+export type OrganizationMembers = OrganizationMember[];
+
+export function transformOrganizationMembersResponse(
+  members: components["schemas"]["Organization"]["members"],
+): OrganizationMembers {
+  return (
+    members?.map((member) => ({
+      memberId: member.id,
```
Comment (mnholtz) [inline src/data/model/OrganizationMembers.ts:46]: This UUID actually refers to the OrganizationMembership id (the primary key for the membership record in the db https://github.com/pixiebrix/pixiebrix-app/blob/fbee39abe61f1b7793066b34487e179ffd426093/api/serializers/account.py#L421-L428)

I think the name `memberId` is misleading in this case, let's maybe call it `membershipId`?

File: src/data/model/OrganizationMembers.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  transformOrganizationMemberUserResponse,
+  type OrganizationMemberUser,
+} from "@/data/model/OrganizationMemberUser";
+import {
+  type UserRoleType,
+  transformUserRoleResponse,
+} from "@/data/model/UserRole";
+import { type components } from "@/types/swagger";
+
+export type OrganizationMember = {
```
Comment (mnholtz) [inline src/data/model/OrganizationMembers.ts:32]: Regarding my comment here https://github.com/pixiebrix/pixiebrix-extension/pull/9152/files#r1759394197 this type is probably better suited to be called `OrganizationMembership`

File: src/data/model/OrganizationMemberUser.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { validateUUID } from "@/types/helpers";
+import { type Timestamp, type UUID } from "@/types/stringTypes";
+import { type components } from "@/types/swagger";
+import { type SetRequired } from "type-fest";
+
+export type OrganizationMemberUser = {
```
Comment (mnholtz) [inline src/data/model/OrganizationMemberUser.ts:23]: Following my suggestion here I think `OrganizationMembershipUser` would be a more appropriate name https://github.com/pixiebrix/pixiebrix-extension/pull/9152/files#r1759397674

File: src/data/model/OrganizationMemberGroups.ts [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { type UUID } from "@/types/stringTypes";
+import { type components } from "@/types/swagger";
+import { type SetRequired } from "type-fest";
+
+export type OrganizationMemberGroup = {
```
Comment (mnholtz) [inline src/data/model/OrganizationMemberGroups.ts:22]: See other comments about moving towards `OrganizationMembership` prefix instead

**Review by mnholtz — COMMENTED** [review]
Ready to approve once changes are made to OrganizationMember -> OrganizationMembership prefix

**Review by mnholtz — APPROVED** [review]

**Pull request merged**